src/coreclr/src/vm/codeversion.h

```diff
@@ -293,6 +307,9 @@ class NativeCodeVersionNode
 #ifdef HAVE_GCCOVER
     PTR_GCCoverageInfo m_gcCover;
 #endif
+#ifdef FEATURE_ON_STACK_REPLACEMENT
+    CORINFO_OSR_INFO m_osrInfo;
```

Comment:
The `CORINFO_PATCHPOINT_INFO` is not DACized (it contains direct pointer). It does not look right to use it as a field inside a fully DACized structure.

I tends to not work very well to use JIT/EE interface structures inside runtime datastructure with longer lifetime. You may want to limit `CORINFO_PATCHPOINT_INFO` to just JIT/EE interface, and use something else here (potentially the offset and the blog pointer without any wrapper).

Reply:
I can make it opaque here, but would want the same definition to be visible to jit and runtime, do we have a good way to do that (utilcode, I guess?).

Reply:
I would make two different structs. It is not unusual for a small amount of duplication to make things a lot simpler.

Reply:
Ah, was thinking of PATCHPOINT_INFO, yes, I can destructure this here.

Reply:
Have revised this, no more CORINFO_OSR_INFO, just passing around (offset, ppinfo) instead.